src/client/useESLint/worker.ts

```diff
@@ -0,0 +1,120 @@
+import * as eslint from 'eslint-linter-browserify';
+import globals from 'globals';
+
+const linter = new eslint.Linter();
+
+// Helper function to trim keys of an object
+function trimGlobalKeys(
+  obj: Record<string, unknown>,
+): Record<string, unknown> {
+  if (!obj || typeof obj !== 'object') {
+    return obj;
+  }
+  const newObj: Record<string, unknown> = {};
+  for (const key in obj) {
+    if (Object.prototype.hasOwnProperty.call(obj, key)) {
+      newObj[key.trim()] = obj[key];
+    }
+  }
+  return newObj;
+}
+
+// Define the configuration without explicit types that don't exist
+const config = {
+  languageOptions: {
+    globals: {
+      ...trimGlobalKeys(globals.browser), // Apply trimming to browser globals
+      ...globals.es2021, // Assuming es2021 globals are fine, or trim them too if needed
+    },
+    parserOptions: {
+      ecmaVersion: 2022, // Use number instead of enum type
+      sourceType: 'module',
+      ecmaFeatures: {
+        // Add this to enable JSX parsing
+        jsx: true,
+      },
+    },
+  },
+  rules: {
+    'no-unused-vars': ['warn', { argsIgnorePattern: '^_' }],
+    'no-undef': 'error',
+    semi: 'off',
+  },
+};
+
+// Helper function to convert line/column to character offset
+function getOffset(
+  docLines: string[],
+  line: number,
+  column: number,
+): number {
+  let offset = 0;
+  for (let i = 0; i < line - 1; i++) {
+    offset += (docLines[i] ? docLines[i].length : 0) + 1; // +1 for newline
+  }
+  return offset + column - 1;
+}
+
+self.onmessage = (event) => {
+  const { code, requestId, fileName } = event.data;
+  if (typeof code !== 'string') {
+    self.postMessage({
+      diagnostics: [],
+      requestId,
+      error: 'Invalid code received',
+    });
+    return;
+  }
+
+  // Only lint .js or .jsx files
+  if (!fileName || !/\.(js|jsx)$/i.test(fileName)) {
```

Comment:
Aha! So this is why the `fileName` is needed. I wonder if we could move this test to be outside of the Worker, where we just don't send the message at all if we are not in .js or .jsx files.